c2c-gateway: refactoring

Danielius1922 opened the pull request:
- update configuration to be loadable from yaml

Danielius1922 committed:
c2c-gateway: refactoring

  - update configuration to be loadable from yaml

Danielius1922 commented on cloud2cloud-gateway/service/config.go:
```diff
@@ -1,31 +1,173 @@
+		return fmt.Errorf("eventBus.%w", err)
+	}
+	if err := c.ResourceAggregate.Validate(); err != nil {
+		return fmt.Errorf("resourceDirectory.%w", err)
```
resourceAggregate

jkralik commented on cloud2cloud-gateway/service/service.go:
```diff
@@ -2,157 +2,291 @@ package service
+
+	subMgr := NewSubscriptionManager(ctx, store, gwClient, cfg.Clients.Subscription.HTTP.ReconnectInterval, emitEvent)
+	if err := subMgr.LoadSubscriptions(); err != nil {
+		return nil, nil, fmt.Errorf("cannot load subscriptions: %w", err)
```
pls, add `fl.Close`

Danielius1922 replied:
Fixed

jkralik commented on cloud2cloud-gateway/config.yaml:
```diff
@@ -0,0 +1,83 @@
+apis:
+  http:
+    address: "0.0.0.0:9100"
+    ownerClaim: "sub"
```
I expected it under `clients.authorizationServer.ownerClaim`

Danielius1922 replied:
It's not needed at all after I've removed the `CtxWithOwner` calls.

Danielius1922 committed:
c2c-gateway: lower cyclomatic complexity of functions

Danielius1922 committed:
resource-aggregate: remove obsolete code

Since c2c-gw was refactored to use new Publisher and Subscriber
structs the original ones were no longer used and thus were
removed.

jkralik commented on cloud2cloud-gateway/service/subscriptions.go:
```diff
@@ -219,39 +271,36 @@ func (s *SubscriptionData) createDeviceSubscription(ctx context.Context, emitEve
+	sub, err := createSubscriptionFunc(ctx, emitEvent, &h)
+	if err != nil {
+		// TODO: ak err je grpc unauthenticated -> cancel subscription
+		// test-case?
```
TODO

Danielius1922 committed:
test/oauth-server: add client id with short timeouts

Danielius1922 committed:
test: add function that setups selected services

Danielius1922 committed:
c2c-gateway: cancel expired subscriptions

During start-up c2c-gw tries to reestablish previously
existing subscriptions. When the access token associated
with the subscription has expired the GRPC server will return
authentication error. When this happens the subscription
data is deleted from the database and cancel subscription
message is sent to the client.
